regression/src/main/java/com/dandb/ecomm/test/pages/creditsignal/CreditSignalPage.java

```
@@ -0,0 +1,97 @@
+import com.dandb.jbehave.BasePage;
+import org.junit.Assert;
+import org.openqa.jetty.html.Break;
+import org.openqa.selenium.*;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+
+import static com.dandb.jbehave.SeleniumConditions.elementToBePresent;
+
+public class CreditSignalPage extends BasePage {
+
+    private org.slf4j.Logger log = org.slf4j.LoggerFactory.getLogger(getClass());
+
+    public CreditSignalPage(WebDriver webDriver) {
+        super(webDriver);
+    }
+
+    @Override
+    public String appUrl() {
+        return appConfig().get(DASHBOARD_APPURL_PROPERTY);
+    }
+
+    @Override
+    public void validate() {
+
+    }
+
+    final String DASHBOARD_APPURL_PROPERTY = "dandb.selenium.dashboard.baseUrl";
+
+    @FindBy(xpath = "//*[contains(text(),'Report as of:')]")
+    private WebElement reportAsOf;
+
+    @FindBy(xpath = "//div[@data-cy='credit_signal_2']")
+    private WebElement creditSignalEntitlement;
+
+    @FindBy(xpath = "//span[@data-cy='creditsignal']")
+    private WebElement expandCreditSignal;
+
+    @FindBy(xpath = "//div[@data-cy='credit_signal_2']//button[contains(text(),'Launch')]")
+    private WebElement creditSignal;
+
+    /**
+     * This function is created to validate that CreditSignal is displayed on the Dashboard
+     * The function will expand the "SHOW ALL" dropdown if the user has CreditSignal for multiple DUNS
+     */
+
+    public CreditSignalPage validateCreditSignalTile() {
+        return (CreditSignalPage) rerunFailedScenario(new BasePage.Steps() {
+            @Override
+            public Object perform(){
+                try {
+                    wait.until(ExpectedConditions.visibilityOf(creditSignalEntitlement));
+                } catch (NotFoundException e) {
+                    System.out.print("User Entitlement Error:  Test failed because the User is not entitled to CreditSignal product. Please change your user. ");
+                }
+
```

Can you please use Assert statement instead of try-catch blocks

---

Example: Assert.assertTrue("The expected success message did not recieved", messageDiv.getText().contains(message));

---

@saihareesu using Assert.assertTrue, who do you printout the customer error message instead of the default assertion failure?

---

Fixed with;
 assertElementVisible(errorMessage "User Entitlement Error: User is not entitled to CreditSignal product.", creditSignalEntitlement);